src/target/riscv/riscv-013.c

```diff
@@ -3405,9 +3408,12 @@ static int read_memory_bus_v1(struct target *target, target_addr_t address,
 
 			const size_t last_key = batch->read_keys_used - 1;
 			for (size_t k = 0; k <= last_key; ++k) {
-				sbvalue[k] = riscv_batch_get_dmi_read_data(batch,
-						last_key - k);
-				buf_set_u32(buffer + i * size + k * 4, 0, 8 * size, sbvalue[k]);
+				sbvalue[k] = riscv_batch_get_dmi_read_data(batch, last_key - k);
+				/* This can be written as: n_bits = MIN(32, 8 * size), but I think
+				 * we'd better to be explicit on how n_bits is calculated */
+				const size_t n_bits = (k != last_key) ? 32 : (8 * size - k * 4 * 8);
```

Comment:
Myself, I think the `MIN(32, 8 * size)` is fine to use.

Reply:
I feel like MIN(32, 8 * size) is quite a hacky way to do this. This is the reason for the comment above. 

I mean:

- if we use MIN () call the code become somewhat difficult to read, since in order to understand it you need to know the surrounding context quite well
- if we use `8 * size - k * 4 * 8` the code becomes a little more verbose, however it becomes quite clear and most readers can understand what is going on.

This looks like a matter of personal preference and not a blocker, am I right? Alternatively I can maybe  re-organize the code and avoid the ternary operator  (by moving the last `buf_set_u32` out of the loop)

Reply:
Thank you for your opinions.

After reading the code, I'd be bit more inclined to use `MIN(32, 8 * size)` but I don't feel strongly. 

Please, feel free to use whichever option. 

(Note: Please don't wait for any further comments from me as I will not be able to return to this one before the start of September.)

Reply:
@MarekVCodasip , @en-sc - could you take a looks once again? I've refactored the code a liittle bit. I use MIN(...) now, but moved the last iteration out of the loop. I think it's marginally better now.

Reply:
@MarekVCodasip jentle ping ...

Reply:
@MarekVCodasip ping once again)

Reply:
Sorry for that.